gatling-datadog/src/main/scala/io/gatling/datadog/DatadogRequests.scala

```diff
@@ -1,249 +1,262 @@
-      }
-    }
+  def sendHistogram(
+      client: Client,
+      metricName: String,
+      dataPoints: Seq[DatadogLabels],
+      tags: Seq[String]
+  ): StatusResponse = {
 
-    finishedUsersRequests.foreach { finishedUsersRequest =>
-      finishedUsersRequest.onComplete {
-        case Success(StatusResponse(status)) =>
-          finishedUsers
-            .keys()
-            .asScala
-            .toSeq
-            .foreach(id => DatadogData.remove(id, finishedUsers))
-        case Failure(e) =>
-          logger.error(s"Finished users request failed with error $e")
-      }
-    }
+    // The specification for Datadog histogram data is given here
+    // https://docs.datadoghq.com/metrics/types/?tab=histogram#metric-types
 
-    requestLatencyRequests.foreach { requestLatencyRequest =>
-      requestLatencyRequest.onComplete {
-        case Success(StatusResponse(status)) =>
-          requestLatency
-            .keys()
-            .asScala
-            .toSeq
-            .foreach(id => DatadogData.remove(id, requestLatency))
-        case Failure(e) =>
-          logger.error(s"Request latency request failed with error $e")
-      }
-    }
+    val values = dataPoints.map(_.value.doubleValue)
 
-    errorCounterRequests.foreach { errorCounterRequest =>
-      errorCounterRequest.onComplete {
-        case Success(StatusResponse(status)) =>
-          errorCounter
-            .keys()
-            .asScala
-            .toSeq
-            .foreach(id => DatadogData.remove(id, errorCounter))
-        case Failure(e) =>
-          logger.error(s"Error count request failed with error $e")
-      }
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.avg",
+      metricType = MetricType.Gauge,
+      value = values.sum / dataPoints.length,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.count",
+      metricType = MetricType.Rate,
+      value = dataPoints.length,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.median",
+      metricType = MetricType.Gauge,
+      value = median(values),
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.max",
+      metricType = MetricType.Gauge,
+      value = values.max,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.min",
+      metricType = MetricType.Gauge,
+      value = values.max,
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.95percentile",
+      metricType = MetricType.Gauge,
+      value = p95(values),
+      tags = tags
+    )
+
+    client.metrics.postSingleMetric(
+      metric = s"$metricName.99percentile",
+      metricType = MetricType.Gauge,
+      value = p99(values),
+      tags = tags
+    )
+  }
+
+  private def median(values: Seq[Double]): Double = {
+    val sortedList = values.sorted
```

Comment:
(nit) lots of these rely on a sorted list, why not sort once and pass into these functions? Can use the same name so it is clear that the expectation is that the list is sorted